hathor/manager.py

```diff
@@ -550,58 +550,76 @@ def _verify_checkpoints(self) -> None:
                     f'hash {block_hash.hex()} was found'
                 )
 
+    def get_new_timestamp(self) -> int:
+        """Generate a timestamp appropriate for a new transaction."""
+        timestamp_now = int(self.reactor.seconds())
+        best_block = self.tx_storage.get_best_block()
+        return max(timestamp_now, best_block.timestamp)
+
```

Comment:
I added this method to help transaction generators pick a consistent timestamp. Because tests don't generate blocks in real-time, there are many cases where the timestamp of the latest block is ahead of the reactor (because blocks need strictly increasing timestamps) and with this PR the `generate_parent_txs` does not support picking tx-parents for _before_ the best block, so many places needed to pick an appropriate timestamp using this exact logic.

Although I've only made use of `get_new_timestamp` in places that are in the error path of some test, not in every place that needs a timestamp for a new transaction (which would be the ideal).